devito/finite_differences/differentiable.py

```diff
@@ -234,17 +235,18 @@ def __new__(cls, obj):
 
     def _doit(obj, args=None):
         cls = diffify._cls(obj)
-        if cls is obj.__class__:
-            return obj
-
         args = args or obj.args
-        newobj = cls(*args, evaluate=False)
 
-        # SymPy objects pretend to be immutable, but in reality they are not.
-        # As facts are deduced, a SymPy core object's ``_assumptions`` data
-        # structure is updated. Since here we use ``evaluate=False`` all over
-        # the place, we have to explicitly reassign the ``_assumptions``.
-        newobj._assumptions = obj._assumptions.copy()
+        if cls is obj.__class__:
+            # Try to just update the args if possible (Add, Mul)
+            try:
+                newobj = obj._new_rawargs(*args, is_commutative=obj.is_commutative)
```

Comment:
so remind me how is `_new_rawargs` "better" or different than `from_args` ? and can you add a comment about why this function and not (for example) from args?

Reply:
also, something's odd here: you create `newobj = ...` but then, line 249, you basically override it? as there's no `return` statement here. What am I missing?

If instead I spotted an error, I suggest to add a test that would have caught this issue

Reply:
>What am I missing?
debugging that weird lone ops error left should be return.

> _new_rawargs "better" or different than from_args ? 
Not much some slight difference on `is_commutative`